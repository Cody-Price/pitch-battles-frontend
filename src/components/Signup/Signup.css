.signup-form {
	width: 400px;
	margin: auto;
}

.firstname-input,
.lastname-input,
.email-input,
.password-input,
.confirmpassword-input {
	height: 20px;
	display: flex;
	padding-bottom: 10px;
}

.student-button {
	cursor: pointer;
	display: inline-flex;
	justify-content: space-between;
}

.teacher-button {
	cursor: pointer;
	display: inline-flex;
	justify-content: space-between;
}

button {
	margin-top: 10px;
	cursor: pointer;
}