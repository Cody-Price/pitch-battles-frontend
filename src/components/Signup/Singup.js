import React, { Component } from 'react';

class Signup extends Component {
	constructor() {
		super();
		this.state = {
			firstName: '',
			lastName: '',
			email: '',
			passwordOne: '',
			passwordTwo: '',
			role: 'student'
		}
	}

	handleChange = (e) => {
		this.setState({[e.target.name]: e.target.value})
	}

	changeRole = (role) => {
		if (role === 'student' && this.state.role !== 'student') {
			this.setState({role})
		} else if (role === 'teacher' && this.state.role !== 'teacher') {
			this.setState({role})
		}
	}

	render() {
		return (
			<section className={`signup-form ${this.props.status}`}>
				<form>
					<p>First Name</p>
					<input type='text' name='firstName' onChange={this.handleChange} value={this.state.firstName} />
					<p>Last Name</p>
					<input type='text' name='lastName' onChange={this.handleChange} value={this.state.lastName} />
					<p>Email</p>
					<input type='email' name='email' onChange={this.handleChange} value={this.state.email} />
					<p>Password</p>
					<input type='password' name='passwordOne' onChange={this.handleChange} value={this.state.passwordOne} />
					<p>Confirm Password</p>
					<input type='password' name='passwordTwo' onChange={this.handleChange} value={this.state.passwordTwo} />
					<div aria-label='choose student role' className={`student-button ${this.state.role}`} onClick={() => this.changeRole('student')}>Student</div>
					<div aria-label='choose teacher role' className={`teacher-button ${this.state.role}`} onClick={() => this.changeRole('teacher')}>Teacher</div>
					<button type='submit'>Sign Up</button>
				</form>
				<button onClick={this.props.landingChange}>Login</button>
			</section>
		)
	}
}

export default Signup;
