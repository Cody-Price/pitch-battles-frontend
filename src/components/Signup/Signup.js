import React, { Component } from 'react';
import './Signup.css';

class Signup extends Component {
	constructor() {
		super();
		this.state = {
			firstName: '',
			lastName: '',
			email: '',
			passwordOne: '',
			passwordTwo: '',
			role: 'student'
		}
	}

	handleChange = (e) => {
		this.setState({[e.target.name]: e.target.value})
	}

	changeRole = (role) => {
		if (role === 'student' && this.state.role !== 'student') {
			this.setState({role})
		} else if (role === 'teacher' && this.state.role !== 'teacher') {
			this.setState({role})
		}
	}

	render() {
		return (
			<section className={`signup-form ${this.props.status}`}>
				<form>
				  <div className='firstname-input'>
					<aria-label>first name</aria-label>
					<input type='text' name='firstName' onChange={this.handleChange} value={this.state.firstName} />
				  </div>
				  <div className='lastname-input'>
					<label>last name</label>
					<input type='text' name='lastName' onChange={this.handleChange} value={this.state.lastName} />
				  </div>
				  <div className='email-input'>
					<label>email</label>
					<input type='email' name='email' onChange={this.handleChange} value={this.state.email} />
				  </div>
				  <div className='password-input'>
					<label>password</label>
					<input type='password' name='passwordOne' onChange={this.handleChange} value={this.state.passwordOne} />
				  </div>
				  <div className='confirmpassword-input'>
					<label>confirm password</label>
					<input type='password' name='passwordTwo' onChange={this.handleChange} value={this.state.passwordTwo} />
				  </div>
					<div aria-label='choose student role' className={`student-button ${this.state.role}`} onClick={() => this.changeRole('student')}>student</div>
					<div aria-label='choose teacher role' className={`teacher-button ${this.state.role}`} onClick={() => this.changeRole('teacher')}>teacher</div>
					<button type='submit'>sign up</button>
				</form>
				<button onClick={this.props.landingChange}>Login</button>
			</section>
		)
	}
}

export default Signup;
