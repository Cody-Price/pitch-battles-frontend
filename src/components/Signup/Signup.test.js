const Enzyme = require('enzyme');
const EnzymeAdapter = require('enzyme-adapter-react-16');

Enzyme.configure({ adapter: new EnzymeAdapter() });
import { shallow } from 'enzyme';
import React from 'react';
import Signup from './Signup';

describe('Signup', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Signup />)
	})

	it('should exist', () => {
		expect(wrapper).toBeDefined();
	})

	it('should render like snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})

	it('should set state on handleChange', () => {
		const event = {
			target: {
				name: 'firstName',
				value: 'Haley'
			}
		}

		wrapper.instance().handleChange(event)
		expect(wrapper.state().firstName).toEqual('Haley')
	})

	it('should set state when changeRole is called', () => {
		const event = {
			target: {
				name: 'role',
				value: 'teacher'
			}
		}

		wrapper.instance().handleChange(event)
		expect(wrapper.state().role).toEqual('teacher')
	})
})