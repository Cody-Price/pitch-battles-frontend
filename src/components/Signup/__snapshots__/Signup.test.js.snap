// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form>
          <p>
            First Name
          </p>
          <input
            name="firstName"
            onChange={[Function]}
            type="text"
            value=""
          />
          <p>
            Last Name
          </p>
          <input
            name="lastName"
            onChange={[Function]}
            type="text"
            value=""
          />
          <p>
            Email
          </p>
          <input
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />
          <p>
            Password
          </p>
          <input
            name="passwordOne"
            onChange={[Function]}
            type="password"
            value=""
          />
          <p>
            Confirm Password
          </p>
          <input
            name="passwordTwo"
            onChange={[Function]}
            type="password"
            value=""
          />
          <div
            aria-label="choose student role"
            className="student-button student"
            onClick={[Function]}
          >
            Student
          </div>
          <div
            aria-label="choose teacher role"
            className="teacher-button student"
            onClick={[Function]}
          >
            Teacher
          </div>
          <button
            type="submit"
          >
            Sign Up
          </button>
        </form>,
        <button>
          Login
        </button>,
      ],
      "className": "signup-form undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              First Name
            </p>,
            <input
              name="firstName"
              onChange={[Function]}
              type="text"
              value=""
            />,
            <p>
              Last Name
            </p>,
            <input
              name="lastName"
              onChange={[Function]}
              type="text"
              value=""
            />,
            <p>
              Email
            </p>,
            <input
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />,
            <p>
              Password
            </p>,
            <input
              name="passwordOne"
              onChange={[Function]}
              type="password"
              value=""
            />,
            <p>
              Confirm Password
            </p>,
            <input
              name="passwordTwo"
              onChange={[Function]}
              type="password"
              value=""
            />,
            <div
              aria-label="choose student role"
              className="student-button student"
              onClick={[Function]}
            >
              Student
            </div>,
            <div
              aria-label="choose teacher role"
              className="teacher-button student"
              onClick={[Function]}
            >
              Teacher
            </div>,
            <button
              type="submit"
            >
              Sign Up
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "First Name",
            },
            "ref": null,
            "rendered": "First Name",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "firstName",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Last Name",
            },
            "ref": null,
            "rendered": "Last Name",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "lastName",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Email",
            },
            "ref": null,
            "rendered": "Email",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Password",
            },
            "ref": null,
            "rendered": "Password",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "passwordOne",
              "onChange": [Function],
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Confirm Password",
            },
            "ref": null,
            "rendered": "Confirm Password",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "passwordTwo",
              "onChange": [Function],
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "choose student role",
              "children": "Student",
              "className": "student-button student",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Student",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "choose teacher role",
              "children": "Teacher",
              "className": "teacher-button student",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Teacher",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form>
            <p>
              First Name
            </p>
            <input
              name="firstName"
              onChange={[Function]}
              type="text"
              value=""
            />
            <p>
              Last Name
            </p>
            <input
              name="lastName"
              onChange={[Function]}
              type="text"
              value=""
            />
            <p>
              Email
            </p>
            <input
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
            <p>
              Password
            </p>
            <input
              name="passwordOne"
              onChange={[Function]}
              type="password"
              value=""
            />
            <p>
              Confirm Password
            </p>
            <input
              name="passwordTwo"
              onChange={[Function]}
              type="password"
              value=""
            />
            <div
              aria-label="choose student role"
              className="student-button student"
              onClick={[Function]}
            >
              Student
            </div>
            <div
              aria-label="choose teacher role"
              className="teacher-button student"
              onClick={[Function]}
            >
              Teacher
            </div>
            <button
              type="submit"
            >
              Sign Up
            </button>
          </form>,
          <button>
            Login
          </button>,
        ],
        "className": "signup-form undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                First Name
              </p>,
              <input
                name="firstName"
                onChange={[Function]}
                type="text"
                value=""
              />,
              <p>
                Last Name
              </p>,
              <input
                name="lastName"
                onChange={[Function]}
                type="text"
                value=""
              />,
              <p>
                Email
              </p>,
              <input
                name="email"
                onChange={[Function]}
                type="email"
                value=""
              />,
              <p>
                Password
              </p>,
              <input
                name="passwordOne"
                onChange={[Function]}
                type="password"
                value=""
              />,
              <p>
                Confirm Password
              </p>,
              <input
                name="passwordTwo"
                onChange={[Function]}
                type="password"
                value=""
              />,
              <div
                aria-label="choose student role"
                className="student-button student"
                onClick={[Function]}
              >
                Student
              </div>,
              <div
                aria-label="choose teacher role"
                className="teacher-button student"
                onClick={[Function]}
              >
                Teacher
              </div>,
              <button
                type="submit"
              >
                Sign Up
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "First Name",
              },
              "ref": null,
              "rendered": "First Name",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "firstName",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Last Name",
              },
              "ref": null,
              "rendered": "Last Name",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "lastName",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password",
              },
              "ref": null,
              "rendered": "Password",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "passwordOne",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Confirm Password",
              },
              "ref": null,
              "rendered": "Confirm Password",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "passwordTwo",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "choose student role",
                "children": "Student",
                "className": "student-button student",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Student",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "choose teacher role",
                "children": "Teacher",
                "className": "teacher-button student",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Teacher",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
